pcm_osx.cpp

```diff
@@ -24,8 +23,8 @@ class PcmOutputMacOSX: public PcmOutput {
     static void callback(void *inUserData, AudioQueueRef inAQ, AudioQueueBufferRef inBuffer);
 };
 
-PcmOutputMacOSX::PcmOutputMacOSX()
- : index(0), aq(NULL)
+PcmOutputMacOSX::PcmOutputMacOSX(uint32_t rate)
+ : PcmOutput(rate), index(0), aq(NULL), sample_rate(rate)
```

Comment:
Does this actually compile? You can't initialize a private parent member in a derived class.